src/Microsoft.Health.Fhir.Api/Modules/FhirModule.cs

```diff
@@ -57,12 +59,28 @@ public void Load(IServiceCollection services)
             services.AddSingleton(xmlParser);
             services.AddSingleton(xmlSerializer);
 
-            services.AddSingleton<IReadOnlyDictionary<ResourceFormat, Func<string, Resource>>>(x =>
+            services.AddSingleton<IReadOnlyDictionary<FhirResourceFormat, Func<string, string, DateTimeOffset, ResourceElement>>>(x =>
```

Comment:
This could be registered with `FhirResourceFormat` and `Func<string, Resource>` right? Since the only difference is that JsonSerializer vs XmlSerializer but the Parse method is the same.

Something like:
```
services.AddSingleton<IReadOnlyDictionary<FhirResourceFormat, Func<string, Resource>>>(x =>
{
    return new Dictionary<FhirResourceFormat, Func<string, Resource>>
    {
        { FhirResourceFormat.Json, jsonParser.Parse<Resource> },
        { FhirResourceFormat.Xml, xmlParser.Parse<Resource> },
    };
});
```

Reply:
Added a local method that sets the common properties